src/codeflare_sdk/cluster/cluster.py

```diff
@@ -583,82 +512,107 @@ def get_current_namespace():  # pragma: no cover
-            appwrapper = _check_aw_exists(cluster_name, namespace)
-            return Cluster.from_k8_cluster_object(
-                rc,
-                appwrapper=appwrapper,
-                write_to_file=write_to_file,
-                verify_tls=verify_tls,
-            )
-    raise FileNotFoundError(
-        f"Cluster {cluster_name} is not found in {namespace} namespace"
+
+def _create_resources(yamls, namespace: str, api_instance: client.CustomObjectsApi):
+    api_instance.create_namespaced_custom_object(
+        group="ray.io",
+        version="v1",
+        namespace=namespace,
+        plural="rayclusters",
+        body=yamls,
     )
 
 
-# private methods
-def _delete_resources(
-    yamls, namespace: str, api_instance: client.CustomObjectsApi, cluster_name: str
+def get_cluster(
+    cluster_name: str,
+    namespace: str = "default",
+    verify_tls: bool = True,
+    is_appwrapper: bool = False,  # As get_cluster no longer searches a list of potential candidates it is necessary to specify if the resource is an appwrapper
+    write_to_file: bool = False,
 ):
-    for resource in yamls:
-        if resource["kind"] == "RayCluster":
-            name = resource["metadata"]["name"]
-            api_instance.delete_namespaced_custom_object(
-                group="ray.io",
-                version="v1",
+    config_check()
+    api_instance = client.CustomObjectsApi(api_config_handler())
+    # Check/Get the AppWrapper if it exists
+    if is_appwrapper:
+        try:
+            resource = api_instance.get_namespaced_custom_object(
+                group="workload.codeflare.dev",
+                version="v1beta2",
                 namespace=namespace,
-                plural="rayclusters",
-                name=name,
+                plural="appwrappers",
+                name=cluster_name,
             )
-
-
-def _create_resources(yamls, namespace: str, api_instance: client.CustomObjectsApi):
-    for resource in yamls:
-        if resource["kind"] == "RayCluster":
-            api_instance.create_namespaced_custom_object(
+        except Exception as e:
+            return _kube_api_error_handling(e)
+    else:
+        # Get the Ray Cluster
+        try:
+            resource = api_instance.get_namespaced_custom_object(
                 group="ray.io",
                 version="v1",
                 namespace=namespace,
                 plural="rayclusters",
-                body=resource,
+                name=cluster_name,
             )
+        except Exception as e:
+            return _kube_api_error_handling(e)
 
+    # Create a Cluster Configuration with just the necessary provided parameters
+    cluster_config = ClusterConfiguration(
+        name=cluster_name,
+        namespace=namespace,
+        verify_tls=verify_tls,
+        write_to_file=write_to_file,
+        appwrapper=is_appwrapper,
+    )
+    cluster = Cluster(cluster_config, is_retrieved_cluster=True)
```

Comment:
Good point I will try to come up with something more elegant. I could just build the _limited_ `ClusterConfiguration` within the `get_cluster()` function rather than alter the existing functionality

Reply:
Can you elaborate a bit more on this? 

That bool exists to prevent a Ray Cluster from being built in `build_ray_cluster.py`. Other wise when we create the _limited_ `ClusterConfiguration` the Cluster object is created but you would get duplicated print statements for `"Written to: {output_file_name}` from first creating the limited cluster then from creating a file from the retrieved cluster.

Reply:
You can create a class function which has different logic which also returns a cluster. Something like:

```
cluster = Cluster.new_cluster_skip_create(cluster_config)
```

Reply:
You are a hero, that is a great idea!

Reply:
@KPostOffice WDYT of the new changes for `get_cluster` I have added? 

When trying to initialise the Cluster object I was getting exception errors for not providing a `ClusterConfiguration`. I tried to think of a way around this and I opted to set the CC as none when using `get_cluster` which would allow me to set the `config` and `resource_yaml` after initialisation. 

I in turn added a warning if a user tried to specify `ClusterConfiguration` as `None`. I feel like this is still a bit crude and would welcome any further suggestions.

Reply:
Do you want to spend some time pair programming this tomorrow? We can spitball some ideas

Reply:
Yeah sounds good I can set up a meeting for later